test/common/http/conn_manager_impl_test.cc

```diff
@@ -2835,6 +2835,136 @@ TEST_F(HttpConnectionManagerImplTest, RequestTimeoutIsDisarmedOnConnectionTermin
   filter_callbacks_.connection_.raiseEvent(Network::ConnectionEvent::RemoteClose);
 }
 
+TEST_F(HttpConnectionManagerImplTest, RequestHeaderTimeoutValidlyConfigured) {
+  request_headers_timeout_ = std::chrono::milliseconds(10);
+  setup(false, "");
+
+  EXPECT_CALL(*codec_, dispatch(_)).WillOnce(Invoke([&](Buffer::Instance&) -> Http::Status {
+    Event::MockTimer* request_header_timer = setUpTimer();
+    EXPECT_CALL(*request_header_timer, enableTimer(request_headers_timeout_, _));
+    EXPECT_CALL(*request_header_timer, disableTimer());
```

Comment:
What triggers this disable?  the request not parsing and triggering an error reply?

Reply:
Yeah that was the actual issue. I made the test more restrictive to verify that the correct behavior is being checked.